### ISSUE TITLE
Cura 5.0

### ISSUE THREAD

**Issue by r00t4rd3d** [bigtreetech/Bigtree3DPluginSuit#18]
How can we get this working with Cura 5.0?

https://github.com/Ultimaker/Cura/releases/tag/5.0-beta

The plugins location has changed to "C:\Program Files\Ultimaker Cura 5.0.0\share\cura\plugins"

I tried putting the existing plugins in that folder but I got an error when starting Cura and it did not work.

**Comment by discip** [comment]
@r00t4rd3d
Thank you for reporting this issue. I am facing the same issue here too.
# The latest official release [Cura 5.0.0](https://github.com/Ultimaker/Cura/releases/tag/5.0.0) is *also* affected!

Unfortunately your statement:

> The plugins location has changed to "C:\Program Files\Ultimaker Cura 5.0.0\share\cura\plugins"

is not true!

Since it is still here: **`C:\Users\<user name>\AppData\Roaming\cura\5.0\plugins`**.

## But the fact, that plugins have been reworked is absolutely true!

@bigtreetech
Please consider updating this highly appreciated plugin too! 🥺

Thanks in advance!
kind regards

**Comment by Guilouz** [comment]
Yes plugins need to be updated to pyQt6 for working with latest Cura release.

**Comment by kalestew** [comment]
Any advice on updating this to pyQt6 myself? This is a much loved plugin for me.

**Comment by BsCmOD** [comment]
Follow

**Comment by Stillbeer** [comment]
I like to see this 3d preview  working in Cura 5.0 !

@bigtreetech 
Please make it work again.

**Comment by killrob** [comment]
intrested too, is too useful this plugin.

**Comment by rstalde** [comment]
Can users expect a version of the plugin adapted to Cura 5.0.X?
I think that the plugin is the reason for many users to pay more than your competitors for the control and display of the 3D printers.
Greetings from the river Rhine.

**Comment by emguepo** [comment]
I back to CURA 4.13.1 👍

**Comment by r00t4rd3d** [comment]
Looks like I am going back to an older version of Cura till this is updated. 

Where the DEV's for this plugin at?

Ive tried to message those involved in the development of this to see if they will update it for us. 

Cross your fingers!

**Comment by DenPride** [comment]
[https://drive.google.com/file/d/11J3-Knk8KsuZX2CsqNBnN9dy3KSaPDDR/view?usp=sharing](url)
If you want the job to be done, do it yourself
Place to  C:\Users \ <user name>\AppData\Roaming\cura\5.0\plugins

**Comment by DenPride** [comment]
Tell my if it doesn't work..... I can't check rignt now

**Comment by discip** [comment]
@DenPride
Thank you for trying to solve this.

Unfortunately you formatted the link the wrong way. 😁
In order to work properly the format should look like this:
```
[BigTree3DPlugin.zip](https://drive.google.com/file/d/11J3-Knk8KsuZX2CsqNBnN9dy3KSaPDDR/view)
```
Result should look like this:
[BigTree3DPlugin.zip](https://drive.google.com/file/d/11J3-Knk8KsuZX2CsqNBnN9dy3KSaPDDR/view)

Second thing is:
The plugin does not work as it did in e.g.: 4.13.1 😭

The only option available is:
![Untitled](https://user-images.githubusercontent.com/53649486/172960838-ff8e7bc7-63a1-4dee-8990-56bd299f19e7.jpg)

The '***save to Removable drive as BigTree3D***' is missing.

Nonetheless thanks for the effort.

kind regards

**Comment by r00t4rd3d** [comment]
It works! 

I used @discip download link and @DenPride instructions and it works for me! Thanks! 

@discip "Save as BigTree3D format" is what you choose and then the save dialog will popup and you can pick where to save.

**Comment by Guilouz** [comment]
Here version compatible with all versions of Cura :

[BigTreeTech 3D Plugin.zip](https://github.com/bigtreetech/Bigtree3DPluginSuit/files/8882316/BigTreeTech.3D.Plugin.zip)

**Comment by emguepo** [comment]
> Here version compatible with all versions of Cura :
> 
> [BigTreeTech 3D Plugin.zip](https://github.com/bigtreetech/Bigtree3DPluginSuit/files/8882316/BigTreeTech.3D.Plugin.zip)

¿This is the same that commented [discip](https://github.com/discip)?

**Comment by emguepo** [comment]
> It works!
> 
> I used @discip download link and @DenPride instructions and it works for me! Thanks!
> 
> @discip "Save as BigTree3D format" is what you choose and then the save dialog will popup and you can pick where to save.

Me too

**Comment by discip** [comment]
Sorry, but am I the only one experiencing this odd behavior of the dialog missing one entry?

In 4.13.1 it looks like this (4 options):
![4 13 1](https://user-images.githubusercontent.com/53649486/173159496-29840abb-fd7c-46ee-9127-fb28364bbe31.jpg)


In 5.0.0 one entry is missing (only 3 options):
![5 0 0](https://user-images.githubusercontent.com/53649486/173159234-61895506-37c5-4fcf-b344-cbd507d779a6.jpg)

thanks in advance

**Comment by emguepo** [comment]
> Sorry, but am I the only one experiencing this odd behavior of the dialog missing one entry?
> 
> In 4.13.1 it looks like this (4 options): ![4 13 1](https://user-images.githubusercontent.com/53649486/173159496-29840abb-fd7c-46ee-9127-fb28364bbe31.jpg)
> 
> In 5.0.0 one entry is missing (only 3 options): ![5 0 0](https://user-images.githubusercontent.com/53649486/173159234-61895506-37c5-4fcf-b344-cbd507d779a6.jpg)
> 
> thanks in advance

No, me too, but in "Save to Removable Drive", we can selected de USB.

**Comment by Guilouz** [comment]
> > Here version compatible with all versions of Cura :
> > [BigTreeTech 3D Plugin.zip](https://github.com/bigtreetech/Bigtree3DPluginSuit/files/8882316/BigTreeTech.3D.Plugin.zip)
> 
> ¿This is the same that commented [discip](https://github.com/discip)?

Nope, this version is compatible with all version of Cura, it’s working with pyQT5 and pyQT6.

**Comment by rstalde** [comment]
The solution works for me with the limitations @discip  already mentioned.
In addition, intermittent problems occurred when overwriting an existing file from Cura 4.12.1 on the SD card. 
The file was not overwritten, only an error message was issued. The file had to be deleted manually.

**Comment by brucky49** [comment]
Mais c'est merveilleux !!!
GOOD JOB 😁

**Comment by Stillbeer** [comment]
Absolutly fantastic!

**Comment by mase1985** [comment]
Hey guys. Thanks for sharing. I still have a problem with the plugin folder for mac - cura 5.1.0 :-) I copied the content of the zip-file to Resources > share > cura > plugins, but it doesn't work, any ideas?

**Comment by ordxpres** [comment]
> [https://drive.google.com/file/d/11J3-Knk8KsuZX2CsqNBnN9dy3KSaPDDR/view?usp=sharing](url) If you want the job to be done, do it yourself Place to C:\Users \ \AppData\Roaming\cura\5.0\plugins

**THANK YOU :-)**

**Comment by discip** [comment]
@bigtreetech
Please update this plugin to work with Cura versions 5.0.0 and higher.

This plugin appears to have been abandoned. 🤷‍♂️

Would you please let the users know what you intend to do with this topic.

Users like @DenPride were able to partially get it to work. (Thanks for that again!)
## But unfortunately it doesn't work like before *(e.g: Cura 4.13.1)*.
### There is no dedicated option to save to a removable drive and 'eject after saving was successful' is gone as well.

Maybe @smartavionics could help us with that? 🥺
If he has time at hand and is willing to do so, that is! 😊

Thanks in advance

**Comment by Guilouz** [comment]
@discip Use my version I posted in this thread. It’s working fine with Cura 5, it’s working also on Cura 5.2 beta

**Comment by discip** [comment]
@Guilouz
I tried that, but as mentioned above, several times, 2 of the previously present options are not available!

**Comment by flatcapper** [comment]
I just downloaded the latest cura and it does prompt to save to removable storage after slicing model.

Sent from Samsung Galaxy smartphone.
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by maquinariasboedo** [comment]
The plugins location has changed to "C:\Program Files\Ultimaker Cura 5.0.0\share\cura\plugins" It is True.

The @discip download link works in that location.

**Comment by sunnyxiong2020** [comment]
@Guilouz 
Tried on Mac version 5.2.1, unfortunately it didn't work

**Comment by Luigisvc** [comment]
I am using 5.2.1 and is working

**Comment by sunnyxiong2020** [comment]
> I am using 5.2.1 and is working

Does it run on Mac or Windows system? I just tried on my Mac it didn’t work

**Comment by Luigisvc** [comment]
> 

I  am using in a Pc with Windows 11

**Comment by RDickens** [comment]
works in cura 5.2.1

**Comment by klap50** [comment]
> > Here version compatible with all versions of Cura :
> > [BigTreeTech 3D Plugin.zip](https://github.com/bigtreetech/Bigtree3DPluginSuit/files/8882316/BigTreeTech.3D.Plugin.zip)
> 
> ¿This is the same that commented [discip](https://github.com/discip)?

That version works very fine in Cura 5.2.1 ! just need drop in new folder  (prgram files --- ultam cura -- shared -- plugins)

**Comment by beffjessen** [comment]
> 

I also can't get this to work on a mac. The only plugins directory I could find was under Ultimaker Cura/Contents/Resources/share/cura/plugins

**Comment by racerty1** [comment]
> > 
> 
> I also can't get this to work on a mac. The only plugins directory I could find was under Ultimaker Cura/Contents/Resources/share/cura/plugins

In windows, if your on Cura 5.2.x open cura, click Help tab up top, then click "show configuration folder"
then navigate to the plugins folder, the base directory layout for the MAC version should be the same as windows

Paste the corrected file (BigeTreeTech 3D Plugin.zip) from @discip above and unzip it in that folder,

**Comment by SebGTx** [comment]
> > Sorry, but am I the only one experiencing this odd behavior of the dialog missing one entry?
> > In 4.13.1 it looks like this (4 options): ![4 13 1](https://user-images.githubusercontent.com/53649486/173159496-29840abb-fd7c-46ee-9127-fb28364bbe31.jpg)
> > In 5.0.0 one entry is missing (only 3 options): ![5 0 0](https://user-images.githubusercontent.com/53649486/173159234-61895506-37c5-4fcf-b344-cbd507d779a6.jpg)
> > thanks in advance
> 
> No, me too, but in "Save to Removable Drive", we can selected de USB.

The problem is in this file [BigTreeRemovableDriveOutputDevice/RemovableDrivePlugin.py](https://github.com/bigtreetech/Bigtree3DPluginSuit/blob/master/BigTreeRemovableDriveOutputDevice/RemovableDrivePlugin.py) in the _addRemoveDrives function that use key already in use by cura, but i don't know how to fix it

**Comment by sunnyxiong2020** [comment]
> Figured out the problem and now it seems working in Mac as well.
The plugin has to be dropped to the folder below(Thanks to "victornpb"

macOS

The default installation path is ~/Library/Application Support/cura/{VERSION}/plugins

Open Finder Go → Go to folder... (⇧ ⌘ G)
Paste ~/Library/Application Support/cura/, then press Enter
Then navigate to version, plugins

**Comment by sunnyxiong2020** [comment]
This is what I tested in Cura 5.4.0, Mac 14.0 Beta
<img width="386" alt="image" src="https://github.com/bigtreetech/Bigtree3DPluginSuit/assets/73984399/f2d981c9-6093-4733-99b8-d451fb817f67">

**Comment by Relixtay** [comment]
![obrazek](https://github.com/bigtreetech/Bigtree3DPluginSuit/assets/60867134/ae635a80-ea71-416a-a5d2-ae8f040f4ef5)

**Comment by fyrak1s** [comment]
Check out #20 it has the fix

**Comment by enagen1** [comment]
> Check out #20 it has the fix

Works perfect!!! Better than the old version. Thanks!
